include/mqtt/property.hpp

```diff
@@ -812,136 +597,42 @@ class retain_available : public detail::n_bytes_property<1> {
 };
 
 
-class user_property;
-
 namespace detail {
 
-struct len_str;
-
-struct len_str_ref {
-    explicit len_str_ref(string_view v)
-        : len{num_to_2bytes(static_cast<std::uint16_t>(v.size()))}
-        , str(as::buffer(v.data(), v.size()))
-    {}
-    len_str_ref(len_str const& v);
-
-    std::size_t size() const {
-        return len.size() + get_size(str);
-    }
-    boost::container::static_vector<char, 2> len;
-    as::const_buffer str;
-};
-
 struct len_str {
```

Comment:
I don't understand what the purpose of this struct is.

boost::container::static_vector<char, 2> len; value can be calculated as needed, on the fly. No need to store it.

The logic from
```
if (!already_checked) {
    auto r = utf8string::validate_contents(buf);
    if (r != utf8string::validation::well_formed) throw utf8string_contents_error(r);
}
```

looks like it could be trivially merged into the UserProperty constructor, couldn't it?

Reply:
Even if we can't eliminate the "len_str" class entirely,

I think it should be moved into the `private:` section of the UserProperty class.

Reply:
I will fix it.

Reply:
`len_str` is moved.